Emby.Server.Implementations/Session/SessionManager.cs

```diff
@@ -508,13 +510,20 @@ private SessionInfo GetSessionInfo(
             ArgumentException.ThrowIfNullOrEmpty(deviceId);
 
             var key = GetSessionKey(appName, deviceId);
+            SessionInfo sessionInfo;
 
-            CheckDisposed();
-
-            if (!_activeConnections.TryGetValue(key, out var sessionInfo))
+            try
+            {
+                _activeConnectionsLock.EnterWriteLock();
+                if (!_activeConnections.TryGetValue(key, out sessionInfo))
+                {
+                    sessionInfo = CreateSession(key, appName, appVersion, deviceId, deviceName, remoteEndPoint, user);
+                    _activeConnections[key] = sessionInfo;
+                }
+            }
```

Comment:
You are not using the Reader part of the lock. I would suggest this pattern instead with [Lock](https://learn.microsoft.com/en-us/dotnet/api/system.threading.lock?view=net-9.0)

```C#
private readonly Lock _activeConnectionsLock = new();
```

```C#
if (!_activeConnections.TryGetValue(key, out sessionInfo))
{
    using (_activeConnectionsLock.EnterScope())
    {
        if (!_activeConnections.TryGetValue(key, out sessionInfo))
        {
            sessionInfo = CreateSession(key, appName, appVersion, deviceId, deviceName, remoteEndPoint, user);
            _activeConnections[key] = sessionInfo;
        }
    }
}
```